components/elm/src/external_models/emi/src/em/alquimia/ExternalModelAlquimiaMod.F90

```diff
@@ -0,0 +1,3019 @@
+          cation_exchange_capacity,&
+          aux_doubles,&
+          aux_ints) 
+
+    porosity_tmp=this%chem_state%porosity
+    call this%chem%ReactionStepOperatorSplit(this%chem_engine, actual_dt, this%chem_properties, this%chem_state, &
+                                         this%chem_aux_data, this%chem_status)
+    ! Reset porosity because Pflotran tends to mess it up
+    this%chem_state%porosity=porosity_tmp
+
+    ! Get auxiliary output
+    call this%chem%getAuxiliaryOutput(this%chem_engine, this%chem_properties, this%chem_state, &
+    this%chem_aux_data, this%chem_aux_output, this%chem_status)
+
+    ! Check for error
+    if(this%chem_status%error /= 0) then
+      call c_f_string_ptr(this%chem_status%message,status_message)
+      call endrun(msg='Alquimia error in getAuxiliaryOutput: '//status_message)
+    endif
+
+    ! In top layer, cut time step if gas species being absorbed very fast because it should be close to equilibrium
+    ! if(j==1) then
+    !   do k=1,this%chem_sizes%num_primary
+    !     if(this%is_dissolved_gas(k) .and. (surf_bc(k) > 0.0) .and. (sat(1)<=0.9) .and. &
+    !         ((surf_bc(k)*porosity(1)*max(sat(1),0.3) - total_mobile(1,k) )/(surf_bc(k)*porosity(1)*max(sat(1),0.3)) > 0.5) &
+    !         .and. max_cuts<4) then
+    !           this%chem_status%converged = .FALSE.
+    !           write(iulog,'(a,f5.2,x,a,x,i4,x,a)'),'Cutting time step to dt = ',actual_dt,' because species',k,'reduced too fast in layer 1'
+    !     endif
+    !   enddo
+    ! endif
+
+
+    if (this%chem_status%converged) then
+      ! Success. Can finish execution of the subroutine
+
+      ! Copy back to column structure
+      call this%copy_Alquimia_to_ELM(j,water_density_tmp,&
+        aqueous_pressure_tmp,&
+        total_mobile_tmp,free_mobile_tmp,&
+        total_immobile_tmp,&
+        mineral_volume_fraction_tmp,&
+        mineral_specific_surface_area_tmp,&
+        surface_site_density_tmp,&
+        cation_exchange_capacity_tmp,&
+        aux_doubles_tmp,&
+        aux_ints_tmp) 
+      
+    else ! Solve did not converge. Cut timestep, and bail out if too short
+      if(actual_dt/2 < min_dt) then
+        call c_f_string_ptr(this%chem_status%message,status_message)
+        ! Sometimes solve fails because time step is too short (not sure why)
+        ! I wonder if at bailout we should first try pausing transport and solving each layer separately?
+        write(msg,'(a,i3,a,f5.3,a,i4,a,i3,a,i5)') "Error: Alquimia ReactionStepOperatorSplit failed to converge after ",num_cuts," cuts to dt = ",actual_dt,' s. Newton iterations = ',this%chem_status%num_newton_iterations,' Layer = ',j!," Col = ",c
+        call print_alquimia_state(this)
+        write(iulog,*),'Alquimia status message: '//status_message
+        call endrun(msg=msg)
+      else
+        exit ! Drop out of the layer loop to start over at shorter time step
+      endif
+    endif
+    enddo ! Layer loop
+
+
+    if(actual_dt<=60.0_r8) then
```

Comment:
I think we should remove this block. I'm not sure it's in the right place and it's usually not necessary